pages/Add/index.js

```diff
@@ -21,43 +19,43 @@ export default function Add({
   setFishLength,
   fishLocation,
   setFishLocation,
-  markerPosition,
-  setMarkerPosition,
+
+  setLatLng,
+  latlng,
+  fetchedData,
 }) {
   const [opened, setOpened] = useState(false);
   const { pathname } = useRouter();
 
-  const handleSubmit = (e) => {
-    e.preventDefault();
-
-    markerPosition.map((marker) => {
-      const newFishItem = {
-        id: nanoid(),
-        fishName: fishName,
-        fishWeight: +fishWeight,
-        fishLength: +fishLength,
+  const sendToServer = async () => {
```

Comment:
There is insufficient validation of your form. The user can type in any destination and it will turn into undefined lng and lat values breaking the app.